base/sparse/sparsevector.jl

```diff
@@ -392,7 +392,7 @@ function Base.getindex{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, i::Integer, J::Abstract
             if rowvalA[ptrA] <= rowI
                 ptrA = searchsortedfirst(rowvalA, rowI, ptrA, stopA, Base.Order.Forward)
                 if ptrA <= stopA && rowvalA[ptrA] == rowI
-                    push!(nzinds, ptrI)
+                    push!(nzinds, j)
```

Comment:
No, it should be `j`.  We want the n-th index to be the result of indexing with to the n-th value in `J`… that just happens to be the same as `col` in the case of `1:end`.